otj-metrics-core/src/main/java/com/opentable/metrics/graphite/Graphite.java

```diff
@@ -0,0 +1,209 @@
+     */
+    public Graphite(String hostname, int port, SocketFactory socketFactory) {
+        this(hostname, port, socketFactory, UTF_8);
+    }
+
+    /**
+     * Creates a new client which connects to the given address and socket factory using the given
+     * character set.
+     *
+     * @param hostname      The hostname of the Carbon server
+     * @param port          The port of the Carbon server
+     * @param socketFactory the socket factory
+     * @param charset       the character set used by the server
+     */
+    public Graphite(String hostname, int port, SocketFactory socketFactory, Charset charset) {
+        this.hostname = hostname;
+        this.port = port;
+        this.address = null;
+        this.socketFactory = socketFactory;
+        this.charset = charset;
+    }
+
+    /**
+     * Creates a new client which connects to the given address using the default
+     * {@link SocketFactory}.
+     *
+     * @param address the address of the Carbon server
+     */
+    public Graphite(InetSocketAddress address) {
+        this(address, SocketFactory.getDefault());
+    }
+
+    /**
+     * Creates a new client which connects to the given address and socket factory.
+     *
+     * @param address       the address of the Carbon server
+     * @param socketFactory the socket factory
+     */
+    public Graphite(InetSocketAddress address, SocketFactory socketFactory) {
+        this(address, socketFactory, UTF_8);
+    }
+
+    /**
+     * Creates a new client which connects to the given address and socket factory using the given
+     * character set.
+     *
+     * @param address       the address of the Carbon server
+     * @param socketFactory the socket factory
+     * @param charset       the character set used by the server
+     */
+    public Graphite(InetSocketAddress address, SocketFactory socketFactory, Charset charset) {
+        this.hostname = null;
+        this.port = -1;
+        this.address = address;
+        this.socketFactory = socketFactory;
+        this.charset = charset;
+    }
+
+    @Override
+    public void connect() throws IllegalStateException, IOException {
+        if (isConnected()) {
+            throw new IllegalStateException("Already connected");
+        }
+        InetSocketAddress address = this.address;
+        if (address == null) {
+            address = new InetSocketAddress(hostname, port);
+        }
+        if (address.getAddress() == null) {
+            // retry lookup, just in case the DNS changed
+            address = new InetSocketAddress(address.getHostName(), address.getPort());
+
+            if (address.getAddress() == null) {
+                throw new UnknownHostException(address.getHostName());
+            }
+        }
+
+        try {
+            this.socket = socketFactory.createSocket(address.getAddress(), address.getPort());
+        } catch (IOException e) {
+            LOGGER.error("PLEASE REPORT TO ARCHTEAM",e);
```

Comment:
Let's us confirm our suspicion

Reply:
Maybe put in the slack channel name as part of the message?